modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/ContentPageResourceImpl.java

```diff
@@ -188,16 +192,34 @@ private Layout _getLayout(long groupId, String friendlyUrlPath)
 		return _layoutLocalService.getLayout(layoutFriendlyURL.getPlid());
 	}
 
-	private SegmentsExperience _getSegmentsExperience(
+	private long _getSegmentsExperienceId(
 			Layout layout, String segmentsExperienceKey)
 		throws Exception {
 
-		if (Validator.isNull(segmentsExperienceKey)) {
-			return null;
+		if (Validator.isNotNull(segmentsExperienceKey)) {
+			SegmentsExperience segmentsExperience =
```

Comment:
@javierdearcos, internally it's a get, so it will throw the exception, even though it's name fetch

Reply:
🤦 I see